include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -1023,4 +1023,27 @@ def TTNN_MeshShardOp: TTNN_Op<"mesh_shard"> {
     let hasVerifier = 1;
 }
 
+def TTNN_PermuteOp : TTNN_Op<"permute"> {
+    let summary = "Permute operation.";
+    let description = [{
+      Permute input tensor dimensions.
+
+      Attributes:
+        - `permutation` array<i64>: The permutation of the input tensor dimensions.
+
+      Example:
+      %a = tensor.empty() : () -> tensor<2x3x4xi32>
+      %0 = "ttir.permute"(%a) {permutation = array<i64: 1, 2, 0>} : (tensor<2x3x4xi32>) -> tensor<3x4x2xi32>
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         DenseI64ArrayAttr:$permutation,
```

Comment:
I'm somewhat torn on what to do here... In the lib, there's a `Span` object, which seems to inherit from boost's span, which seems to be yet another container for objects. I'd like to differentiate it from a `DenseI64ArrayAttr` so that the RT doesn't need to "know" about this (the overloaded methods for processing operands talk...).

Reply:
I don't see a problem, `DenseI64ArrayAttr` is MLIRs equivalent of `std::vector`, `span` on the other hand is just a view over array-like object, similar to `ArrayRef` in LLVM, so if we always convert `DenseI64ArrayAttr` to `std::vector` it's going to work out of the box.

Reply:
I did not know that, that's wonderful, thank you! 😊